packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,242 @@
+                    activeId={this.state.activeId}
+                    onUpdate={this.handleUpdate}
+                    items={this.slides}
+                    managedClasses={this.generateTabsClassNames()}
+                />
+                <Flipper
+                    direction={FlipperDirection.next}
+                    onClick={this.handleNextClick}
+                    className={get(this.props, "managedClasses.carousel_flipperNext", "")}
+                />
+            </div>
+        );
+    }
+
+    /**
+     * Generate class names
+     */
+    protected generateClassNames(): string {
+        let className: string = get(this.props, "managedClasses.carousel", "");
+        let theme: string = "";
+
+        if (this.getSlideTheme()) {
+            theme =
+                this.getSlideTheme() === SlideTheme.light
+                    ? get(this.props, "managedClasses.carousel__themeLight", "")
+                    : get(this.props, "managedClasses.carousel__themeDark", "");
+
+            className += ` ${theme}`;
+        }
+
+        return super.generateClassNames(className);
+    }
+
+    /**
+     * Returns tabs managedclasses with new carousel-specific JSS
+     */
+    protected generateTabsClassNames(): TabsClassNameContract {
+        return {
+            tabs: get(this.props, "managedClasses.carousel_slides", ""),
+            tabs_tabPanels: get(this.props, "managedClasses.carousel_tabPanels", ""),
+            tabs_tabList: get(
+                this.props,
+                "managedClasses.carousel_sequenceIndicators",
+                ""
+            ),
+            tabs_tabPanelContent: get(
+                this.props,
+                "managedClasses.carousel_tabPanelContent",
+                ""
+            ),
+            tab: get(this.props, "managedClasses.carousel_sequenceIndicator", ""),
+            tab__active: get(
+                this.props,
+                "managedClasses.carousel_sequenceIndicator__active",
+                ""
+            ),
+            tabPanel: get(this.props, "managedClasses.carousel_tabPanel", ""),
+            tabPanel__hidden: get(
+                this.props,
+                "managedClasses.carousel_tabPanel__hidden",
+                ""
+            ),
+        };
+    }
+
+    /**
+     * Get the active slide index
+     */
+    private getActiveIndex(): number {
+        return this.slides.map((slide: Slide) => slide.id).indexOf(this.state.activeId);
+    }
+
+    /**
+     * Get the current slide theme (needed for UI overlay contrast)
+     */
+    private getSlideTheme(): string {
+        if (!this.slides || this.slides.length <= 0) {
+            return;
+        }
+
+        return this.slides[this.getActiveIndex()].theme
+            ? this.slides[this.getActiveIndex()].theme
+            : SlideTheme.dark;
+    }
+
+    /**
+     * Updates slides
+     */
+    private updateSlides(): void {
+        this.slides = this.props.items.slice();
```

Comment:
This will throw if `this.props.items` is not an array. I think you should remove this and change your `slides` prop to the following:

```
private get slides(): Slide[] {
    return Array.isArray(this.props.items) ? this.props.items : [];
}
```

That will allow you to safely access slides without checking to ensure it's an array everywhere you need to access slides, and also means you don't have to keep a private member in sync with props.

Reply:
Awesome. Makes sense - updated.